tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -394,5 +450,19 @@ public void TestConvertObjectIdToObjectId()
 
             Assert.Equal(oid, oidConverted);
         }
+
+        // nested types
+        private class AppDomainCourier : MarshalByRefObject
+        {
+            public ObjectId ObjectId => ObjectId.GenerateNewId();
```

Comment:
Let's use a method instead of a property to avoid any confusion about whether the property is serialized or not between app domains.

    public ObjectId CreateObjectId() => ObjectId.GenerateNewId();

Reply:
done